Tools/refcounts/.ruff.toml

```diff
@@ -0,0 +1,33 @@
+target-version = "py312"
+line-length = 79
+fix = true
+preview = true
+
+[lint]
+select = [
+    "I", # isort
+    "F", # Enable all pyflakes rules
+    "E", # Enable all pycodestyle error rules
+    "W", # Enable all pycodestyle warning rules
+    "UP", # Enable all pyupgrade rules
+    "RUF100", # Ban unused `# noqa` comments
+    "C4", # flake8-comprehensions
+    "ISC", # flake8-implicit-str-concat
+    "TCH", # flake8-type-checking
+    "PTH", # flake8-use-pathlib
+    "PERF", # perflint
```

Comment:
heh, I feel like the first one of these two categories emphasises readability and correctness, sometimes at the cost of performance; the second emphasises micro-optimisations, sometimes at the cost of readability and correctness

Reply:
Do you mean "PTH" would be for correctness but may be slow while the intent of "PERF" is to have better performances at the cost of correctness? To me, the "PERF" rules are also for readability so they don't conflict with each others.

Reply:
> Do you mean "PTH" would be for correctness but may be slow while the intent of "PERF" is to have better performances at the cost of readability/correctness?

yes

> To me, the "PERF" rules are also for readability so they don't conflict with each others.

It depends on the rule. Fixing PERF203 often demands changes that, while more performant, would be considered by many people to be unidiomatic or at least unusual. See e.g. https://github.com/astral-sh/ruff/pull/12947

Reply:
I've selected the rules that I wanted (namely for loops instead of comprehensions or .copy()).